pkg/datasources/system_get_aws_sns_iam_policy.go

```diff
@@ -30,7 +31,7 @@ var systemGetAWSSNSIAMPolicySchema = map[string]*schema.Schema{
 
 func SystemGetAWSSNSIAMPolicy() *schema.Resource {
 	return &schema.Resource{
-		ReadContext: TrackingReadWrapper(datasources.SystemGetAwsSnsIamPolicy, ReadSystemGetAWSSNSIAMPolicy),
+		ReadContext: PreviewFeatureReadWrapper(string(previewfeatures.SystemGetAWSSNSIAMPolicyDatasource), TrackingReadWrapper(datasources.SystemGetAwsSnsIamPolicy, ReadSystemGetAWSSNSIAMPolicy)),
```

Comment:
nit: SystemGetAWSSNSIAMPolicyDatasource would be nicer as SystemGetAwsSnsIamPolicyDatasource - I know that golang prefers uppercased abbreviations but in such cases it's just wrong and unreadable - in our code, In multiple places we do a first-letter-only uppercasing